lib/chef/resource/aws_rds_instance.rb

```diff
@@ -28,12 +28,21 @@ class Chef::Resource::AwsRdsInstance < Chef::Provisioning::AWSDriver::AWSRDSReso
   attribute :additional_options, kind_of: Hash, default: {}
 
   def aws_object
-    res = driver.rds.instances[name]
-    if res.exists? && ! ['deleted', 'deleting'].include?(res.status)
-      res
-    else
-      nil
-    end
+    result = self.driver.rds_resource.db_instance(name)
+    return nil unless result && result.db_instance_status != 'deleting'
+    result
+  rescue ::Aws::RDS::Errors::DBInstanceNotFound
+    nil
+  end
+
+  def endpoint
```

Comment:
The only method needed to get at the backing AWS object is the `aws_object` method on the Resource.  When I do a `ctrl+f` on this page, I do not use any reference to `endpoint` being called.  Thats why I asked whether this method was needed.  `db_instance_status` is used elsewhere so I understand having that method.

We disable resource cloning for chef-provisioning resources, so that isn't an issue.